src/platform/wayland.rs

```diff
@@ -24,10 +24,10 @@ pub trait ActiveEventLoopExtWayland {
     fn is_wayland(&self) -> bool;
 }
 
-impl ActiveEventLoopExtWayland for ActiveEventLoop {
+impl ActiveEventLoopExtWayland for &dyn ActiveEventLoop {
     #[inline]
     fn is_wayland(&self) -> bool {
-        self.p.is_wayland()
+        self.as_any().downcast_ref::<crate::platform_impl::wayland::ActiveEventLoop>().is_some()
```

Comment:
In general, though, such APIs should be added to `ActiveEventLoop` with `-> Option<dyn &Extension>`, but for now I just show how you can implement them un-conditionally with the help of casting, etc.